agent/command/git.go

```diff
@@ -870,15 +871,34 @@ func (c *gitFetchProject) fetch(ctx context.Context,
 		}
 	}
 
-	// Clone the project's modules.
+	// For every module, expand the module prefix.
 	for _, moduleName := range conf.BuildVariant.Modules {
-		if err := ctx.Err(); err != nil {
-			return errors.Wrapf(err, "canceled while applying module '%s'", moduleName)
-		}
-		err = c.fetchModuleSource(ctx, comm, conf, logger, jpm, td, opts.token, opts.method, p, moduleName)
+		module, err := conf.Project.GetModuleByName(moduleName)
 		if err != nil {
-			return errors.Wrapf(err, "fetching module source '%s'", moduleName)
+			return errors.Wrapf(err, "getting module '%s'", moduleName)
+		}
+		if module == nil {
+			return errors.Errorf("module '%s' not found", moduleName)
 		}
+		expandModulePrefix(conf, module.Name, module.Prefix, logger)
```

Comment:
expandModulePrefix returns a string, in addition to adding something to the conf list; if we're purposefully ignoring the return value, I think _ = expandModulePrefix(...) is the right notation to use, to indicate this.

Reply:
Also it doesn' tseem like a problem to be doing this in a loop here as opposed to in fetchModuleSource directly, but I think I don't fully understand why we're doing this here instead.

Reply:
I can add a comment but expanding the module prefixes sets a value in a map field inside of `conf`. Setting a value in a map in a gorountine is not concurrent safe- it's only concurrent safe to read from a map concurrently, not write.

The helper function is used somewhere else where it does use the returned string, I'll look in either splitting the function or consolidating the return/usage to be consistent in both places.